tests/Integration/WorkloadManagementSystem/Test_TaskQueueDB.py

```diff
@@ -1,152 +1,355 @@
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  # this should match whatever
+  assert int(result['Value'][0][0]) in [tq_job1, tq_job2, tq_job3,
+                                        tq_job4, tq_job5, tq_job6]
+  assert len(result['Value']) == 5
+
+  # slc5 -- this time it should match 1 (the one without specified platform)
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'slc5'},
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  assert int(result['Value'][0][0]) == tq_job6
+  assert len(result['Value']) == 1
+
+  # slc6
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'slc6'},
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  assert int(result['Value'][0][0]) in [tq_job4, tq_job6]
+  assert len(result['Value']) == 2
+
+  # Now we insert a TQ with platform "ANY" (same as no platform)
+
+  tqDefDict = {'OwnerDN': '/my/DN', 'OwnerGroup': 'myGroup', 'Setup': 'aSetup', 'CPUTime': 5000,
+               'Platform': 'ANY'}
+  result = tqDB.insertJob(7, tqDefDict, 10)
+  assert result['OK'] is True
+  result = tqDB.getTaskQueueForJobs([7])
+  tq_job7 = result['Value'][7]
+  assert tq_job7 == tq_job6  # would be inserted in the same TQ
+
+  # matching for this one
+
+  # ANY
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'ANY'},
+                                     numQueuesToGet=6)
+  assert result['OK'] is True
+  # this should match whatever
+  assert int(result['Value'][0][0]) in [tq_job1, tq_job2, tq_job3,
+                                        tq_job4, tq_job5, tq_job6, tq_job7]
+  assert len(result['Value']) == 5
+
+  # slc5 -- this time it should match 2
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'slc5'},
+                                     numQueuesToGet=6)
+  assert result['OK'] is True
+  assert int(result['Value'][0][0]) in [tq_job6, tq_job7]
+  assert len(result['Value']) == 1
+
+  # slc6
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'slc6'},
+                                     numQueuesToGet=6)
+  assert result['OK'] is True
+  assert int(result['Value'][0][0]) in [tq_job4, tq_job6, tq_job7]
+  assert len(result['Value']) == 2
+
+  # new platform appears
+  # centos8 (> centos7)
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Platform': 'centos8'},
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  # FIXME: here, I would like to see 3 TQs matched: those for slc6 + centos7 + ANY
```

Comment:
Let me know what you think of this

Reply:
I do not understand the relation centos8 > centos7. Do you mean that on centos8 platform jobs requiring centos7 can be executed ? Where this rlation is formally specified ? in OSCompatibility settings ? This looks very confusing to me.